### PR TITLE
auth: Rewrite OpenSSLRSADNSCryptoKeyEngine::fromISCMap to be less tricky

### PR DESCRIPTION
**Pull request by omoerbeek**
### Short description

Prompted by CodeQL

### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [X] tested this code
- [ ] included documentation (including possible behaviour changes)
- [x] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

File: pdns/opensslsigners.cc [hunk 1]
```diff
@@ -429,88 +430,44 @@ std::string OpenSSLRSADNSCryptoKeyEngine::getPublicKeyString() const
 }
 
 
+std::unique_ptr<BIGNUM, void(*)(BIGNUM*)>OpenSSLRSADNSCryptoKeyEngine::parse(std::map<std::string, std::string>& stormap, const std::string& key)
+{
+  const std::string& v = stormap[key];
+  auto n = std::unique_ptr<BIGNUM, void(*)(BIGNUM*)>(BN_bin2bn((const unsigned char*)v.c_str(), v.length(), nullptr), BN_clear_free);
```
Comment (rgacogne) [inline pdns/opensslsigners.cc:436]:
```suggestion
  auto n = std::unique_ptr<BIGNUM, void(*)(BIGNUM*)>(BN_bin2bn(reinterpret_cast<const unsigned char*>(v.data()), v.length(), nullptr), BN_clear_free);
```

**Review by rgacogne — APPROVED** [review]

File: pdns/opensslsigners.cc [hunk 2]
```diff
@@ -429,88 +430,44 @@ std::string OpenSSLRSADNSCryptoKeyEngine::getPublicKeyString() const
 }
 
 
+std::unique_ptr<BIGNUM, void(*)(BIGNUM*)>OpenSSLRSADNSCryptoKeyEngine::parse(std::map<std::string, std::string>& stormap, const std::string& key)
+{
+  const std::string& v = stormap[key];
```
[the remark this answers is not in the document]
Comment (omoerbeek) [reply pdns/opensslsigners.cc:435]: If absent, the choice is exception on getting the value or on parsing an empty string

File: pdns/opensslsigners.cc (hunk 1 above)
Comment (omoerbeek) [reply pdns/opensslsigners.cc:436]: Any particular reason on `data()` vs `c_str()`? In C++11, they are equivalent, AFAIK.

Comment (rgacogne) [reply pdns/opensslsigners.cc:436]: In my mind `c_str()` implies that we need a null-terminated string, which is not true here. But I don't really care either way :)

**Pull request merged**